card-unification.md

```diff
@@ -0,0 +1,90 @@
+# Card Unification
+
+  * [ ] Change card shape to object with `type`, `name`, `render` and optional `edit` properties
+  * [ ] Card's `type` is validated by renderer (dom renderer cannot render 'text', e.g.)
+  * [ ] Change arguments passed by editor to card's `render` (or `edit`) method
+    * [ ] single argument object with `name`, `buffer`, `isInEditor`, `options`, `editorEnv` and `payload` properties
+  * [ ] Return value of card's `render` (or `edit`) method is appended/concatenated by renderer
+  * [ ] card can register teardown callback by calling `buffer.onTeardown(teardownFn)`
+  * [ ] Change editor-dom renderer to clear child elements from card element on teardown
+  * [ ] Renderer `constructor` signature changes to accept cards (array), cardOptions (object) arguments
```

Comment:
`editor` is only necessary for the editor-dom renderer (there is no editor when using the other renderers).

`unknownCardHandler` needs to be accounted for. Currently only the editor-dom renderer has any concept of it. We should codify it for all renderers. I'll update this accordingly.

Given that a renderer may (optional) require an editor, and will need `cards`, `cardAtoms`, `cardOptions` and `unknownCardHandler` I am inclined to change the signature of the renderer constructor to accept an `options` argument instead of this increasing list of positional arguments.